lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -58,20 +65,36 @@ export class AuthorizationCodeClient extends BaseClient {
             this.logger
         );
 
+        // Validate response. This function throws a server error if an error is returned by the server.
         responseHandler.validateTokenResponse(response.body);
-        const tokenResponse = responseHandler.generateAuthenticationResult(response.body, this.defaultAuthority);
+        const tokenResponse = responseHandler.generateAuthenticationResult(response.body, this.authority, cachedNonce, cachedState);
```

Comment:
Can be done in a separate PR but at some point we should add this to BaseClient - These are used and duplicate across all client at the moment.

Reply:
the nonce and state checks are specific to this flow though.